dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,529 @@
+    {
+        Verify.NotNullOrWhiteSpace(key);
+
+        return this.DeleteItemsAsync([key], cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public Task DeleteBatchAsync(IEnumerable<string> keys, DeleteRecordOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(keys);
+
+        return this.DeleteItemsAsync(keys, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public Task DeleteCollectionAsync(CancellationToken cancellationToken = default)
+    {
+        return this.RunOperationAsync("DeleteContainer", () =>
+            this._database
+                .GetContainer(this.CollectionName)
+                .DeleteContainerAsync(cancellationToken: cancellationToken));
+    }
+
+    /// <inheritdoc />
+    public async Task<TRecord?> GetAsync(string key, GetRecordOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(key);
+
+        return await this.GetBatchAsync([key], options, cancellationToken)
+            .FirstOrDefaultAsync(cancellationToken)
+            .ConfigureAwait(false);
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<TRecord> GetBatchAsync(
+        IEnumerable<string> keys,
+        GetRecordOptions? options = null,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(keys);
+
+        const string OperationName = "GetItems";
+
+        var includeVectors = options?.IncludeVectors ?? false;
+        var fields = new List<string>(includeVectors ? this._storagePropertyNames.Values : this._nonVectorStoragePropertyNames);
+        var queryDefinition = this.GetSelectQuery(keys.ToList(), fields);
+
+        await foreach (var jsonObject in this.GetItemsAsync(queryDefinition, cancellationToken).ConfigureAwait(false))
+        {
+            yield return VectorStoreErrorHandler.RunModelConversion(
+                DatabaseName,
+                this.CollectionName,
+                OperationName,
+                () => this._mapper.MapFromStorageToDataModel(jsonObject, new() { IncludeVectors = includeVectors }));
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<string> UpsertAsync(TRecord record, UpsertRecordOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(record);
+
+        const string OperationName = "UpsertItem";
+
+        var jsonObject = VectorStoreErrorHandler.RunModelConversion(
+                DatabaseName,
+                this.CollectionName,
+                OperationName,
+                () => this._mapper.MapFromDataToStorageModel(record));
+
+        var jsonObjectKey = jsonObject.TryGetPropertyValue(this._keyStoragePropertyName, out var jsonKey) ? jsonKey?.ToString() : null;
+        var jsonObjectPartitionKeyValue = jsonObject.TryGetPropertyValue(this._partitionKeyStoragePropertyName, out var jsonPartitionKey) ? jsonPartitionKey?.ToString() : null;
```

Comment:
Do we need to check in the constructor if the partition key property type is a string, so that we know this won't fail later?

Reply:
Good idea! Added a validation for partition key.